src/agent/main.cpp

```diff
@@ -124,6 +127,10 @@ static int Mainloop(otbr::AgentInstance &aInstance, const char *aInterfaceName)
     RestWebServer *restServer = RestWebServer::GetRestWebServer(&ncpOpenThread);
     restServer->Init();
 #endif
+#ifdef OTBR_ENABLE_VENDOR_BUS_SERVER
+    otbrVendorBusInit(ncpOpenThread);
```

Comment:
Should we follow the style of `dbusAgent` and `restServer` which use C++ methods of an object but not `C` style functions? e.g.:
```c++
#if OTBR_ENABLE_VENDOR_BUS_SERVER
VendorBusServer *vendorBus = VendorBusServer::GetVendorBusServer(&ncpOpenThread);
vendorBus->Init();
#endif
```

And this `VendorBusServer` can inherit `otbr::MainloopProcessor`. Thoughts?